code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
@tractorcow In my testing it worked fine and allows me to use Versioned::current_stage() just fine.
(Returning "Stage" when the session is set to Stage, and returning "Live" otherwise)

So I'm not sure if you've actually tested this or just speculating.